src/train_gui.cpp

```diff
@@ -512,19 +512,52 @@ void DrawTrainDetails(const Train *v, int left, int right, int y, int vscroll_po
 		CargoArray act_cargo;
 		CargoArray max_cargo;
 		Money feeder_share = 0;
+		uint16 empty_weight = 0;
+		uint16 loaded_weight = 0;
+		uint16 empty_max_speed = 0;
+		uint16 loaded_max_speed = 0;
 
 		for (const Vehicle *u = v; u != nullptr; u = u->Next()) {
 			act_cargo[u->cargo_type] += u->cargo.StoredCount();
 			max_cargo[u->cargo_type] += u->cargo_cap;
 			feeder_share             += u->cargo.FeederShare();
+			empty_weight             += Train::From(u)->GetEmptyWeight();
+			loaded_weight            += Train::From(u)->GetLoadedWeight();
 		}
 
-		/* draw total cargo tab */
-		DrawString(left, right, y + text_y_offset, STR_VEHICLE_DETAILS_TRAIN_TOTAL_CAPACITY_TEXT);
-		y += line_height;
+		const auto rolling_friction = 15 * (512 + v->GetDisplayMaxSpeed()) / 512;
+		const auto tractive_effort_empty = empty_weight * rolling_friction;
+		const auto tractive_effort_loaded = loaded_weight * rolling_friction;
+		const auto power = v->gcache.cached_power * 746ll;
+		const auto max_te = v->gcache.cached_max_te;
+		empty_max_speed = std::min<uint16>(v->GetDisplayMaxSpeed(), (tractive_effort_empty == 0 || tractive_effort_empty > max_te) ? 0 : (3.6 * (power / tractive_effort_empty)));
+		loaded_max_speed = std::min<uint16>(v->GetDisplayMaxSpeed(), (tractive_effort_loaded == 0 || tractive_effort_loaded > max_te) ? 0 : (3.6 * (power / tractive_effort_loaded)));
```

Comment:
Have you looked into resolving this compiler warning, so that I don't have to include it in #267 when the PR gets merged?

Warning: `comparison of integers of different signs: 'const int' and 'const unsigned int' [-Wsign-compare]`
In other words: `comparison of integer expressions of different signedness: ‘const int’ and ‘const unsigned int’ [-Wsign-compare]`
Detected on: Emscripten, MacOS, Linux

Reply:
Should be fixed :)